zio-schema/shared/src/main/scala/zio/schema/validation/Predicate.scala

```diff
@@ -69,4 +70,27 @@ object Predicate {
   final case class True[A]() extends Predicate[A] { // A => True
     def validate(value: A): Result = Right(Chunk.empty)
   }
+
+  final case class Optional[A](pred: Bool[Predicate[A]], validNone: Boolean) extends Predicate[Option[A]] {
+
+    def validate(value: Option[A]): Result = value match {
+      case None =>
+        if (validNone) Right(Chunk(ValidationError.EqualToNone())) else Left(Chunk(ValidationError.EqualToNone()))
+      case Some(v) => Validation(pred).validate(v).map(_ => Chunk.empty)
+    }
+  }
+
+  final case class Contramap[B, A](pred: Bool[Predicate[A]], f: (B => A)) extends Predicate[B] {
```

Comment:
Storing a Scala function inside `Contramap` may complicate the story of converting a validation into a JSON Spec. Is contramap strictly required?